### DISCUSSION TITLE
Different team colors between the APIs and the plotting module

### DISCUSSION THREAD

**Discussion by pesaventofilippo** [theOehrly/Fast-F1#419] [General]
The team colors in `fastf1.plotting.TEAM_COLORS` are different from the ones returned from the API, for example:

`print(fastf1.plotting.team_color("Ferrari"))  # "#dc0000"`
`print(session.get_driver("LEC").TeamColor)    # "#F91536"`

I think the colors returned by the API are more "pleasant", although the ones in the plotting module are more different from one another and maybe more readable when plotted together.

Is this the reason the two sources don't match? If so, would it be useful to add a parameter to the `plotting.team_color()` function so that you can choose which of the two "palettes" to use?

**Comment by theOehrly** [comment]
I actually never thought about this before or compared them. The primary team color is chose by me. I use a primary color from the teams website or similar usually. Although I noticed just now that the current primary color on mercedes website is slightly different for example. Maybe it changed, I'm not sure. 
In any case, the color that is returned by the API is different yet again. They might be using more "pleasant" colors, as you put it. Where the colors are less vibrant so that they aren't standing out too much. 

Using the color palette from the API would require a separate API request (with a cached result) directly in the plotting module. This could be added in the future. Although I don't see this as a very important thing to add. What exactly would be the use case for this?

**Comment by pesaventofilippo** [comment]
> In any case, the color that is returned by the API is different yet again.

> Using the color palette from the API would require a separate API request

I didn't think the colors returned by the API would change (more frequently than every year?)
If that's the case, then I would argue it's not worth to have them if you have to call the APIs just to have different colors.
If instead they change only every year, you could simply hard-code them in plotting.py (for example, `plotting.API_TEAM_COLORS` alongside `plotting.TEAM_COLORS`), and then add an optional keyword parameter in `plotting.team_color()` to choose between the two palettes.

> Although I don't see this as a very important thing to add. What exactly would be the use case for this?

Yes, it would only be a nice-to-have. For example, if you wanted to plot every team against each other the colors used in the plotting module are more visible and different, but if you need to write a driver's name using their team color, then having the colors stand out is much less important, and the colors from the API work nicer when used on text (specifically on dark backgrounds).
But again, it's only a nice to have, so it's not worth to have them if they'd need an API call and caching.

**Reply by theOehrly** [reply]
I agree, with both. This just needs to be added reasonably to not confuse users why there are similar but different colors. I'll look into it at some point.
